buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt

```diff
@@ -464,36 +256,137 @@ class PerformanceTestPlugin : Plugin<Project> {
+            description = "Runs ad-hoc performance tests on $testProject - can be used locally"
+            channel = "adhoc"
+            outputs.doNotCacheIf("Is adhoc performance test") { true }
+            mustRunAfter(currentlyRegisteredTestProjects)
+
+            retry {
+                maxRetries.set(0)
+            }
+        })
+
+        val channelSuffix = if (OperatingSystem.current().isLinux) "" else "-${OperatingSystem.current().familyName.toLowerCase()}"
+
+        registeredPerformanceTests.add(createPerformanceTest("${testProject}PerformanceTest", generatorTask) {
+            description = "Runs performance tests on $testProject - supposed to be used on CI"
+            channel = "commits$channelSuffix"
+
+            retry {
+                maxRetries.set(1)
+            }
+
+            if (shouldLoadScenariosFromFile) {
+                val scenariosFromFile = project.loadScenariosFromFile(testProject)
+                if (scenariosFromFile.isNotEmpty()) {
+                    setTestNameIncludePatterns(scenariosFromFile)
+                }
+                doFirst {
+                    assert((filter as DefaultTestFilter).includePatterns.isNotEmpty()) { "Running $name requires to add a test filter" }
+                }
+            }
+            mustRunAfter(currentlyRegisteredTestProjects)
+        })
+        registeredTestProjects.add(generatorTask)
+        return generatorTask
+    }
+
+    private
+    fun createPerformanceTest(name: String, generatorTask: TaskProvider<out Task>, configure: PerformanceTest.() -> Unit = {}): TaskProvider<out PerformanceTest> {
+        val performanceTest = project.tasks.register(name, PerformanceTest::class) {
+            group = "verification"
+            buildId = System.getenv("BUILD_ID")
+            reportDir = project.layout.buildDirectory.file("${this.name}/${Config.performanceTestReportsDir}").get().asFile
+            resultsJson = project.layout.buildDirectory.file("${this.name}/${Config.performanceTestResultsJson}").get().asFile
+            addDatabaseParameters(project.propertiesForPerformanceDb())
+            testClassesDirs = performanceSourceSet.output.classesDirs
+            classpath = performanceSourceSet.runtimeClasspath
+
+            performanceTestService.set(buildService)
+
+            testProjectName.set(generatorTask.name)
+            testProjectFiles.from(generatorTask)
+
+            val identityExtension = project.the<ModuleIdentityExtension>()
+            val gradleBuildBranch = identityExtension.gradleBuildBranch.get()
+            branchName = gradleBuildBranch
+            systemProperty("gradleBuildBranch", gradleBuildBranch)
+            commitId.set(identityExtension.gradleBuildCommitId)
+
+            reportGeneratorClass.set("org.gradle.performance.results.report.DefaultReportGenerator")
 
+            maxParallelForks = 1
+            useJUnitPlatform()
+            jvmArgs("-Xmx5g", "-XX:+HeapDumpOnOutOfMemoryError")
```

Comment:
Any idea why the test worker is so large? Is this because of IDEA/Android Studio perf tests?

Reply:
Good question. I don't think the Android Studio TAPI call should use that much memory on the client. I'll try in a different PR with a lower setting. How about `1G`?

Reply:
See https://github.com/gradle/gradle/pull/14728.